### DISCUSSION TITLE
Cosmopolitan run anywhere binary

### DISCUSSION THREAD

**Discussion by iskrisis** [libvips/libvips#3804] [Q&A]
This is probably very stupid question but could one use [Cosmopolitan](https://justine.lol/cosmopolitan/) to build one Vips binary? Has anyone tried this? I obviously know very little i just started to use some cosmopolitan libs for myself because i have to switch architectures and its very pleasant.

**Comment by jcupitt** [comment]
Hi @iskrisis,

I'd not seen cosmo, it looks interesting. I don't think we'd use it, but of course anyone downstream would be very welcome to try. It ought to be simple, I'd think.

The downside of things like this is that for something like libvips, which has a lot of large optional dependencies, the binary could be pretty chunky, perhaps 30mb (just a guess). The hefty size would probably make it rather slow compared to a native binary.

**Reply by iskrisis** [reply]
@jcupitt thank you for your answer! This is more like a shoutout to someone who knows their way around build tools etc. who would try this. I eventually will try to do it myself but i am typical stupid webdeveloper and its a bit over my head. But i already use Cosmopolitan binaries as very crude "same environment" between architectures for development. It's probably not good for production but for development it helps with everyone having same dependencies in project with no issues. And since we use vips i thought it might be good fit.

Anyway cheers and thank you kindly for your time and work @jcupitt
